homeassistant/components/motionmount/coordinator.py

```diff
@@ -0,0 +1,40 @@
+"""Update coordinator for the MotionMount."""
+import logging
+from typing import Any
+
+import motionmount
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MotionMountCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Coordinator for MotionMount."""
+
+    _attr_always_update = False
+
+    def __init__(self, hass: HomeAssistant, mm: motionmount.MotionMount) -> None:
+        """Initialize the MotionMount coordinator."""
+        super().__init__(hass, _LOGGER, name="MotionMount")
+        self._mm = mm
+
+    @property
+    def mm(self) -> motionmount.MotionMount:
+        """Returns the MotionMount."""
+        return self._mm
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        return {
+            "extension": self._mm.extension,
+            "turn": self._mm.turn,
+            "is_moving": self._mm.is_moving,
+            "target_extension": self._mm.target_extension,
+            "target_turn": self._mm.target_turn,
+            "error_status": self._mm.error_status,
+        }
+
+    def motionmount_callback(self) -> None:
+        """Update data from updated MotionMount."""
+        self.hass.add_job(self.async_refresh)
```

Comment:
Is the callback from another thread or is it running in the event loop?

Reply:
The callback is called from the event loop. The library is fully build on asyncio and does not (actively) create any threads.

Reply:
in this case use `self.hass.async_create_task`

Reply:
Actually since you already have the data you can use `self.async_set_updated_data` but I'm wondering why even use a `DataUpdateCoordinator` at all since no data is being polled here AFAICT

Reply:
Ooh, this is an interesting one

Reply:
I used the DataCoordinator as the documentation mentions its use even with a pushing model, so I thought it to be best practice. At the time I wrote that code only a single listener could be registered to the MotionMount library, which is no longer the case so this coordinator-less construct makes sense now. 
I'll adapt it tomorrow.